src/main/java/ChattyCharlie.java

```diff
@@ -0,0 +1,189 @@
+import java.util.Scanner;
+
+public class ChattyCharlie {
+
+    //TASK CLASS
+    public static class Task{
+        protected String description;
+        protected boolean isDone;
+
+        //constructor
+        public Task(String description) {
+            this.description = description;
+            this.isDone = false;
+        }
+
+        //check if its marked as done
+        public String getMarkedStatus() {
+            return (isDone ? "X" : " ");
+        }
+
+        //to toggle the task
+        public void markTask() {
+            this.isDone = true; //change the variable
+        }
+
+        public void unmarkTask() {
+            this.isDone = false; //change the variable
+        }
+    }
+
+
+    //LIST CLASS
+    public static class List {
+        //make a list of task
+        private Task[] list;
+        private int size;
+
+        //constructor
+        public List() {
+            list = new Task[100];
+            size = 0;
+        }
+
+        //Method to add an item to the list
+        public void addTask(String text) {
+            //create an instance for Task
+            Task newTask = new Task(text);
+            //add the text into the list
+            list[size] = newTask;
+            //account for the item
+            size++;
+        }
+
+        //To mark
+        public void mark(int index) {
+            if (index >= 0 && index < size) {
+                list[index].markTask();
+                int remainingTask = countUnmarkedTasks();
+                System.out.println("        Well Done! 1 task down, " + remainingTask + " to go.");
```

Comment:
Perhaps the space identation could be made into a constant variable given its usage throughout the code.